pkg/destroy/bootstrap/bootstrap.go

```diff
@@ -43,20 +39,24 @@ func Destroy(dir string) (err error) {
 	}
 	defer os.RemoveAll(tempDir)
 
-	for _, filename := range copyNames {
+	for _, filename := range []string{
+		terraform.StateFileName,
+		cluster.TfVarsFileName,
+		"disable-bootstrap-load-balancer-targets.tfvars",
+	} {
 		err = copy(filepath.Join(dir, filename), filepath.Join(tempDir, filename))
 		if err != nil {
 			return errors.Wrapf(err, "failed to copy %s to the temporary directory", filename)
 		}
 	}
 
-	if platform == "libvirt" {
-		_, err = terraform.Apply(tempDir, platform, fmt.Sprintf("-var-file=%s", filepath.Join(tempDir, "disable-bootstrap.tfvars")))
-		if err != nil {
-			return errors.Wrap(err, "Terraform apply")
-		}
+	_, err = terraform.Apply(tempDir, platform, fmt.Sprintf("-var-file=%s", filepath.Join(tempDir, "disable-bootstrap-load-balancer-targets.tfvars")))
+	if err != nil {
+		return errors.Wrap(err, "Terraform apply")
 	}
 
+	time.Sleep(10 * time.Second)
```

Comment:
What is this?

Reply:
> What is this?

This is

3. Wait 10 seconds for requests to the bootstrap machine to drain out.

from my topic post.

Reply:
From [the AWS network load-balancer docs][1]:

> Connection draining ensures that in-flight requests complete before existing connections are closed. The initial state of a deregistering target is draining. By default, the state of a deregistering target changes to unused after 300 seconds. To change the amount of time that Elastic Load Balancing waits before changing the state to unused, update the deregistration delay value. We recommend that you specify a value of at least 120 seconds to ensure that requests are completed.

And from [here][2]:

> Deregistering a target removes it from your target group, but does not affect the target otherwise. The load balancer stops routing requests to a target as soon as it is deregistered. The target enters the draining state until in-flight requests have completed.

The Terraform attachment-deletion logic is [here][3], and while it fires a deregister request, it does not wait around for draining to complete.  I don't see any issues in the provider repository about waiting for the unused state, but we could push something like that if we wanted more finesse here than a 10-second cross-platform sleep.  I'm also fine just saying "we know who our consumers are at this point, and none of them will keep an open request going for more than 10 seconds", which is how I have it now.  Thoughts?

[1]: https://docs.aws.amazon.com/elasticloadbalancing/latest/network/target-group-health-checks.html
[2]: https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-target-groups.html#registered-targets
[3]: https://github.com/openshift/installer/blob/v0.11.0/pkg/terraform/exec/plugins/vendor/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_lb_target_group_attachment.go#L80-L106

Reply:
I've pushed 5ee7650 -> a349ed1 working the above into the commit message.